commands/fcp/addlog.js

```diff
@@ -1,69 +1,114 @@
+          option
+              .setName('log-reason')
+              .setDescription('Log content')
+              .setRequired(true)
+      ),
+
   async execute(interaction) {
-    try {
-      const user = interaction.options.getUser("user");
-      const logReason = interaction.options.getString("log-reason");
+    if (
+        interaction.member.roles.cache.some(role => role.name === 'Staff') ||
+        interaction.member.roles.cache.some(role => role.name === 'Air Marshals')
+    ) {
+      try {
+        const user = interaction.options.getUser("user");
+        const logReason = interaction.options.getString("log-reason");
 
-      const response = await fetch(`${Config.fcpBaseUrl()}/api/users/find/${user.id}/audit-logs/new`, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-          'Accept': 'application/json',
-          'Authorization': `Bearer ${process.env.FCP_TOKEN}`
-        },
-        body: JSON.stringify({
-          userId: user.id,
-          staffId: interaction.user.id,
-          text: logReason,
-        }),
-      });
+        const addLogUrl = `${Config.fcpBaseUrl()}/api/users/find/${user.id}/audit-logs/new`;
+
+        const addAuditLog = async () => {
+          return await fetch(addLogUrl, {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json',
+              'Accept': 'application/json',
+              'Authorization': `Bearer ${process.env.FCP_TOKEN}`,
+            },
+            body: JSON.stringify({
+              'user_id': user.id,
+              'submitted_by': interaction.user.id,
+              'text': logReason,
+            }),
+          });
+        };
+
+        let response = await addAuditLog();
+
+        if (response.status === 422) {
 
-      if (response.status === 404) {
-        // TODO: Create a FCP account if user doesn't exist
-      } else if (!response.ok) {
-        throw new Error(`HTTP error. Status: ${response.status}`);
-      } else {
-        const body = await response.json();
+          const createUserUrl = `${Config.fcpBaseUrl()}/api/users/new`;
+
+          const createUserResponse = await fetch(createUserUrl, {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json',
+              'Accept': 'application/json',
+              'Authorization': `Bearer ${process.env.FCP_TOKEN}`,
+            },
+            body: JSON.stringify({
+              'id': user.id,
+            }),
+          });
+
+          if (!createUserResponse.ok) {
+            const errorText = await createUserResponse.text();
+            throw new Error(`Failed to create FCP account. Status: ${createUserResponse.status}, Response: ${errorText}`);
+          }
+          response = await addAuditLog();
+
+          if (response.status === 422) {
+            throw new Error(`Failed to add audit log after creating user. Status: ${response.status}`);
```

Comment:
This needs to be sent to sentry or something so we can track it without having to look at argo